packages/component/src/providers/CustomElements/private/useUpdater.tsx

```diff
@@ -0,0 +1,87 @@
+import { useCallback, useMemo, useRef, type DependencyList } from 'react';
+
+export function createElementRegistryWithUpdater<T extends Element = HTMLElement>() {
+  type disposeFn = void | (() => void);
+  type UpdateFn = (element: T) => disposeFn;
+
+  class ElementUpdater<T> {
+    element: any;
+    #update: UpdateFn;
+    #dispose: disposeFn;
+
+    constructor(element: T, updateFn: UpdateFn) {
+      this.element = element;
+      this.#update = updateFn;
+      this.update();
+    }
+
+    update() {
+      this.dispose();
+      this.#dispose = this.#update(this.element);
+    }
+
+    dispose() {
+      this.#dispose && this.#dispose();
+      this.#dispose = undefined;
+    }
+
+    eqUpdate(updateFn: UpdateFn) {
+      return this.#update === updateFn;
+    }
+  }
+
+  const updateCallbacks = new Set<UpdateFn>();
+  const elements = new Set<T>();
+
+  let updaters = new Set<ElementUpdater<T>>();
+
+  function useUpdater(updateFn: UpdateFn, deps: DependencyList) {
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const updateCallback = useCallback(updateFn, deps);
+
+    const disposeCallback = useRef<() => void>();
```

Comment:
Could this be `useRefFrom`?

Reply:
I'd be happy to stack the two, but it doesn't work well, but `useMemo` executes the callback immediately when no ref yet created. So you'll get:
```
ReferenceError: Cannot access 'disposeCallback' before initialization
```